src/Bicep.Core/Parser/Parser.cs

```diff
@@ -720,17 +786,19 @@ private SkippedTriviaSyntax SynchronizeAndReturnTrivia(int startReaderPosition,
             var skippedSpan = TextSpan.SafeBetween(skippedTokens, startToken.Span.Position);
             var errorSpan = skippedSpan;
 
-            if (consumeTerminator)
+            if (flags.HasFlag(RecoveryFlags.ConsumeTerminator))
             {
                 Synchronize(true, expectedTypes);
 
                 skippedTokens = reader.Slice(startReaderPosition, reader.Position - startReaderPosition);
                 skippedSpan = TextSpan.SafeBetween(skippedTokens, startToken.Span.Position);
             }
 
-            var error = errorFunc(DiagnosticBuilder.ForPosition(errorSpan));
+            var errors = flags.HasFlag(RecoveryFlags.SuppressDiagnostics)
```

Comment:
Are there ever any cases where this could cover up errors without generating new ones? Any way to enforce that contract in code?

Reply:
If called incorrectly yes.

The alternative I was considering was having the the `ParseDiagnosticVisitor` suppress parse errors that are stacked one after another on the same zero-length span. The upside is that the approach simplies the parser logic a bit. The downside is that it moves that logic elsewhere and also introduces the concept of NOT collecting all the errors that we have in the tree. (Collecting parse errors at different parts of the tree would result in a different set of errors, which seems not ideal.)